src/PAModel/IR/IRStateHelpers.cs

```diff
@@ -294,7 +301,10 @@ private static (ControlInfoJson.Item item, int index) CombineIRAndState(BlockNod
                             if (arg.Identifier == PAConstants.ThisPropertyIdentifier)
                                 continue;
 
-                            properties.Add(GetPropertyEntry(state, errors, funcName + "_" + arg.Identifier, arg.Default.Expression));
+                            // Use InvariantScript expression if present otherwise use Default expression.
+                            var propName = funcName + "_" + arg.Identifier;
+                            var script = entropy.FunctionParamsInvariantScripts.ContainsKey(propName) ? entropy.FunctionParamsInvariantScripts[propName] : arg.Default.Expression;
```

Comment:
>entropy.FunctionParamsInvariantScripts [](http://example.com/codeflow?start=41&length=38)

nit: can we use GetOrDefault()  ? 


entropy.FunctionParamsInvariantScripts.GetOrDefault(propName, arg.Default.Expression);